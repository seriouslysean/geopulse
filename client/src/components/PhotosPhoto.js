import React from "react";

const PhotosPhoto = ({ photo }) => {
  const {
    id,
    caption: { text: alt = "" },
    images: { standard_resolution: image },
    likes,
    link,
    user: { username }
  } = photo;
  return (
    <li className="photos__photo">
      <a rel="external" href={link} title={alt}>
        <div className="photos__image" style={{ backgroundImage: `url(${image.url})` }}>
          <img src={image.url} width={image.width} height={image.height} alt={alt} />
        </div>
        <div className="photos__username">{username}</div>
      </a>
    </li>
  );
};

export default PhotosPhoto;
