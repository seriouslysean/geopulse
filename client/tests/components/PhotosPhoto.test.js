import React from 'react';
import { shallow } from 'enzyme';
import PhotosPhoto from '../../src/components/PhotosPhoto';
import photosProp from '../../../data/fixtures/photos';

test('should render PhotosPhoto with content correctly', () => {
  const wrapper = shallow(<PhotosPhoto photo={photosProp.collection[0]} />);
  expect(wrapper).toMatchSnapshot();
});
