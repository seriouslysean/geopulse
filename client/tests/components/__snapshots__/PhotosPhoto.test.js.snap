// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render PhotosPhoto with content correctly 1`] = `
<li
  className="photos__photo"
>
  <a
    href="https://www.instagram.com/p/BckAiH6lMsI/"
    rel="external"
    title="Itâ€™s getting extra Christmasy this morning, hahaha. #ðŸŽ„ #captaincrunch"
  >
    <div
      className="photos__image"
      style={
        Object {
          "backgroundImage": "url(https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/25012870_1945812425670287_2388775529164570624_n.jpg)",
        }
      }
    >
      <img
        alt="Itâ€™s getting extra Christmasy this morning, hahaha. #ðŸŽ„ #captaincrunch"
        height={640}
        src="https://scontent.cdninstagram.com/t51.2885-15/s640x640/sh0.08/e35/25012870_1945812425670287_2388775529164570624_n.jpg"
        width={640}
      />
    </div>
    <div
      className="photos__username"
    >
      seriouslysean
    </div>
  </a>
</li>
`;
